From ffe7825f62c9c565b450b31690565e333ceb6c53 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@intel.com>
Date: Tue, 8 Jan 2019 10:40:34 +0000
Subject: [linux-steam-integration 30/31] Properly set preload libs in conf

Weird janky escaping is left-over from our time as an
autotools project and should've been fixed long ago as
it was obvious the compiler warnings were legit.

Signed-off-by: Ikey Doherty <ikey.doherty@intel.com>
---
 meson.build       | 4 +++-
 meson_options.txt | 2 +-
 src/shim/shim.c   | 4 ++--
 3 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index ee621f2..8f3c28f 100644
--- a/meson.build
+++ b/meson.build
@@ -49,7 +49,8 @@ cdata.set_quoted('LOCALEDIR', path_localedir)
 cdata.set_quoted('GETTEXT_PACKAGE', meson.project_name())
 
 # Configurables
-cdata.set_quoted('LSI_PRELOAD_LIBS', get_option('with-preload-libs'))
+with_preload_libs = get_option('with-preload-libs')
+cdata.set_quoted('LSI_PRELOAD_LIBS', with_preload_libs)
 cdata.set_quoted('STEAM_BINARY', get_option('with-steam-binary'))
 
 # Future reports
@@ -67,6 +68,7 @@ if shim_behaviour != 'none'
     shim_enabled = true
 endif
 shim_system_report += '    behaviour:                              @0@'.format(shim_behaviour)
+shim_system_report += '    preload libraries:                      @0@'.format(with_preload_libs)
 
 # Do we have liblsi-intercept?
 with_libintercept = get_option('with-libintercept')
diff --git a/meson_options.txt b/meson_options.txt
index f915cca..6c16a58 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -3,7 +3,7 @@ option('with-shim', type : 'combo', choices : ['replacement', 'co-exist', 'none'
 option('with-steam-binary', type: 'string', description: 'Set the path to the Steam binary', value: '/usr/lib/steam/steam')
 option('with-frontend', type: 'boolean', description: 'Build the LSI Settings UI', value: false)
 option('with-preload-libs', type: 'string', description: 'Colon separated list of libraries required to launch Steam games',
-        value: '/usr/\$LIB/libxcb.so.1:/usr/\$LIB/libX11.so.6:/usr/\$LIB/libstdc++.so.6')
+        value: '/usr/$LIB/libxcb.so.1:/usr/$LIB/libX11.so.6:/usr/$LIB/libstdc++.so.6')
 option('with-libintercept', type: 'boolean', description: 'Enable libintercept library for controlling Steam linking', value: true)
 option('with-libredirect', type: 'boolean', description: 'Enable libredirect library for fine-tuning Steam games', value: true)
 option('with-new-libcxx-abi', type: 'boolean', description: 'Enable when using the new libstdc++ ABI', value: false)
diff --git a/src/shim/shim.c b/src/shim/shim.c
index a7ddce7..ff4d882 100644
--- a/src/shim/shim.c
+++ b/src/shim/shim.c
@@ -31,14 +31,14 @@
  * Audit path is used for the libintercept library to ensure Steam only uses the
  * host SDL, etc.
  */
-#define AUDIT_PATH "/usr/\$LIB/liblsi-intercept.so"
+#define AUDIT_PATH "/usr/$LIB/liblsi-intercept.so"
 
 /**
  * Redirect path is used for the libredirect library to perform internal
  * redirections in functions like `open()` where needed to hotfix games
  * at runtime.
  */
-#define REDIRECT_PATH "/usr/\$LIB/liblsi-redirect.so"
+#define REDIRECT_PATH "/usr/$LIB/liblsi-redirect.so"
 
 /**
  * Potential for the host Vulkan ICD files (old vs new)
-- 
2.20.0

